clients/admin-ui/src/features/common/custom-fields/CustomFieldsList.tsx

```diff
@@ -1,169 +1,41 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import { Center, Divider, Flex, Spinner, Text } from "@fidesui/react";
-import { FieldArray, useFormikContext } from "formik";
-import { useCallback, useEffect, useMemo, useState } from "react";
 
-import { CustomSelect } from "~/features/common/form/inputs";
-import { useAlert } from "~/features/common/hooks";
-import {
-  useGetAllAllowListQuery,
-  useGetCustomFieldDefinitionsByResourceTypeQuery,
-  useGetCustomFieldsForResourceQuery,
-  useUpsertCustomFieldMutation,
-} from "~/features/plus/plus.slice";
-import { AllowedTypes, ResourceTypes } from "~/types/api";
+import { ResourceTypes } from "~/types/api";
 
-import { useFeatures } from "../features";
+import { CustomFieldSelector } from "./CustomFieldSelector";
 import { CustomFieldsModal } from "./CustomFieldsModal";
-import { CustomFieldWithIdExtended } from "./types";
-
-const initialValues = {
-  customFields: [] as CustomFieldWithIdExtended[],
-};
-
-type FormValues = typeof initialValues;
+import { useCustomFields } from "./hooks";
 
 type CustomFieldsListProps = {
-  resourceId: string;
+  resourceFidesKey?: string;
   resourceType: ResourceTypes;
 };
 
-const CustomFieldsList = ({
-  resourceId,
+export const CustomFieldsList = ({
+  resourceFidesKey,
   resourceType,
 }: CustomFieldsListProps) => {
-  const { errorAlert, successAlert } = useAlert();
-
-  const { flags, plus } = useFeatures();
-  const skip = !(flags.customFields && plus);
-  const [isLoading, setIsLoading] = useState(!skip);
-
-  const { data: allAllowList } = useGetAllAllowListQuery(true, { skip });
-  const { data: customFieldDefinitions } =
-    useGetCustomFieldDefinitionsByResourceTypeQuery(resourceType, { skip });
-  const { data: customFields } = useGetCustomFieldsForResourceQuery(
-    resourceId,
-    { skip }
-  );
-  const [upsertCustomField] = useUpsertCustomFieldMutation();
-
-  // Reference parent Formik context
   const {
-    isSubmitting,
-    setSubmitting,
-    values: formValues,
-  } = useFormikContext();
-
-  const editCustomFieldList = useMemo(() => {
```

Comment:
The logic of these memoized bits has been extracted to a hook: `useCustomFields`. It's also been restructured, which I'll comment on that file.